### ISSUE TITLE
Bean Count QC analysis in SeSame?

### ISSUE THREAD

**Issue by WalterPavicic** [zwdzwd/sesame#153]
Hi,
Some paper analyzing EPIC data (ex: https://doi.org/10.1186/s13148-023-01459-z), when running QC analysis also check for "Probe/Bead Counts" and so: "Probes with more than 5% of samples having a low bead number (< 3) were removed"

For Minfi, one package doing this is: "**beadcount**: Creates matrix of beacounts from minfi data."
Creates matrix of beacounts from minfi data. Description: Creates matrix of beacounts from data read in using the minfi package. NAs represent probes with beadcount <3. An Extended RG Channel Set is required for this function to work.

In Sesame, there is no mention about this "bead count" nor in the pdf guideline (March 16th, 2024: https://www.bioconductor.org/packages/devel/bioc/manuals/sesame/man/sesame.pdf), neither in the SeSame html doc.
Also, in the lastest article by Kaur, D. et al.  (https://doi.org/10.1186/s43682-023-00021-5), this bead count analysis is not mentioned.

So, is Sesame running, or not, this QC in the **prepSesame functions**?

finally, as a general question, should be run/take into account this Bead Count or not?

 thanks, Walter

**Comment by zwdzwd** [comment]
Thank you for bringing this to our attention. We have not considered bead counts in sesame. A quick implementation adds a min_beads option in openSesame (unfortunately, this bead level information is not stored in the standard SigDF object. So you would need to use `openSesame(..., min_beads=4)` to do the filter. Currently the min_beads default is set to 1 to be backward compatible. This will be incorporated in the next release but installable in devel (version 1.21.15+) here: 
https://github.com/zwdzwd/sesame/blob/1a92c6461b26207987c5ae722639d6fcf01febc4/R/open.R#L105
I actually don't have first-hand experience with bead count. But it seems to make sense conceptually to consider low bead counts to be unreliable. And it seems it's a small number of masking (~1000) when min_beads = 4.

**Comment by WalterPavicic** [comment]
Hi @zwdzwd!
Thanks for your quick reply.
Ok, it's interesting to know that bead count will be part of SeSame in the upcoming releases.
Currently Im in sesame: 1.21.13
so will update to latest devel version.

So, if I got it right I can run the bead count, for example when getting betas by using the standard code:
> betas = openSesame("path_to_idats", BPPARAM = BiocParallel::MulticoreParam(2))

Im still in the learning stages, so...where should I add the code: "min_beads=4"
something like: > betas = openSesame("path_to_idats", min_beads=4, ...) ???

Only a few articles mention correction with bead counts, most do not mention having used it.
but when it is mentioned they indicate bead count <3; did you indicate <4 based on your expertise with EPIC data?
Im working on EPICv2

thanks for your kindly support!

**Comment by zwdzwd** [comment]
Yes, your command was correct. I think min_beads = 4 is equivalent to <3 removal but my experience with this filter is quite limited.

**Comment by WalterPavicic** [comment]
Good to know the command is correct.
I´ll update to latest devel and check how it work and what is the number of masking probes.
thanks for your kindly support!

PD: out of this topic...but, is the "InferSex function" working for a set of samples= *.idats? of I need to run one sample at a time?

**Comment by zwdzwd** [comment]
Np! Re inferSex: One sample at a time, but you could always lapply or mclapply.

**Comment by WalterPavicic** [comment]
Hi @zwdzwd 
Just a brief update on this topic...

I´ve now installed latest devel SeSame version
R: 4.4.0
Bioconductor: 3.19
sesame: 1.21.15
sesameData: 1.21.10
ExperimentHub: 2.11.1

To check the new beadcount function, I quickly studied a set of 8 samples (=8 *.idats).
Running both commands:
1) betas_idats_BeadCount4 <- openSesame(idats, min_beads=4, BPPARAM = BiocParallel::SnowParam(2))

2) betas_idats_QCDPB <- openSesame(idats, BPPARAM = BiocParallel::SnowParam(2))

I could not see any differences in total number of NAs.
Then I tested another set of 4 samples, and the same result, NA with and without running beadcount=4 gives same final number.

Of course the answer could simple be that no extra probes were masked by the bead count function...
Data is from EPICv2.
Now I´m going to check the full set of samples I have (N=63)

Edit. PD: I also checked each of the 12 studies samples one at a time.
> sum(is.na(betas2_BC4[,1]))
[1] 69240
> sum(is.na(betas2_QCDPB[,1]))
[1] 69240

best
Walter
